### PR TITLE
Add autoscaler metrics

### PR DESCRIPTION
**Pull request by bjormel**
I confirm that this contribution is made under the terms of the license found in the root directory of this repository's source tree and that I have the authority necessary to make this contribution on behalf of its copyright owner.

### CODE REVIEW

File: metrics/src/main/java/ai/vespa/metrics/ConfigServerMetrics.java [hunk 1]
```diff
@@ -147,7 +147,14 @@ public enum ConfigServerMetrics implements VespaMetrics {
...
     CLUSTER_AUTOSCALED("clusterAutoscaled", Unit.OPERATION, "Number of times a cluster has been rescaled by the autoscaler"),
-
+    CLUSTER_AUTOSCALE_IDEAL_LOAD_CPU("clusterAutoscale.idealLoad.cpu", Unit.VCPU, "The ideal CPU load of a cluster"),
+    CLUSTER_AUTOSCALE_PEAK_LOAD_CPU("clusterAutoscale.peakLoad.cpu", Unit.VCPU, "The CPU load adjustment of a cluster"),
+    CLUSTER_AUTOSCALE_IDEAL_LOAD_MEMORY("clusterAutoscale.idealLoad.memory", Unit.GIGABYTE, "The ideal memory load of a cluster"),
+    CLUSTER_AUTOSCALE_PEAK_LOAD_MEMORY("clusterAutoscale.peakLoad.memory", Unit.GIGABYTE, "The memory load adjustment of a cluster"),
+    CLUSTER_AUTOSCALE_IDEAL_LOAD_DISK("clusterAutoscale.idealLoad.disk", Unit.GIGABYTE, "The ideal disk load of a cluster"),
+    CLUSTER_AUTOSCALE_PEAK_LOAD_DISK("clusterAutoscale.peakLoad.disk", Unit.GIGABYTE, "The disk load adjustment of a cluster"),
+    CLUSTER_AUTOSCALE_IDEAL_LOAD_GPU("clusterAutoscale.idealLoad.gpu", Unit.VCPU, "The ideal GPU load of a cluster"),
+    CLUSTER_AUTOSCALE_PEAK_LOAD_GPU("clusterAutoscale.peakLoad.gpu", Unit.VCPU, "The GPU load adjustment of a cluster"),
```
Comment (mpolden) [inline metrics/src/main/java/ai/vespa/metrics/ConfigServerMetrics.java:157]: I think the unit should be `Unit.FRACTION` for these.

Comment (bratseth) [reply metrics/src/main/java/ai/vespa/metrics/ConfigServerMetrics.java:157]: "The [unit] load adjustment of a cluster" -> "The peak [unit] load observed in the autoscaling window"

File: node-repository/src/main/java/com/yahoo/vespa/hosted/provision/autoscale/Autoscaler.java [hunk 2]
```diff
@@ -69,6 +74,17 @@ public Autoscaling autoscale(Application application, Cluster cluster, NodeList
...
+
+        var context = metric.createContext(dimensions(application.id(), clusterNodes.clusterSpec()));
+        metric.add(ConfigServerMetrics.CLUSTER_AUTOSCALE_IDEAL_LOAD_CPU.baseName(), model.idealLoad().cpu(), context);
+        metric.add(ConfigServerMetrics.CLUSTER_AUTOSCALE_PEAK_LOAD_CPU.baseName(), model.peakLoad().cpu(), context);
+        metric.add(ConfigServerMetrics.CLUSTER_AUTOSCALE_IDEAL_LOAD_MEMORY.baseName(), model.idealLoad().memory(), context);
+        metric.add(ConfigServerMetrics.CLUSTER_AUTOSCALE_PEAK_LOAD_MEMORY.baseName(), model.peakLoad().memory(), context);
+        metric.add(ConfigServerMetrics.CLUSTER_AUTOSCALE_IDEAL_LOAD_DISK.baseName(), model.idealLoad().disk(), context);
+        metric.add(ConfigServerMetrics.CLUSTER_AUTOSCALE_PEAK_LOAD_DISK.baseName(), model.peakLoad().disk(), context);
+        metric.add(ConfigServerMetrics.CLUSTER_AUTOSCALE_IDEAL_LOAD_GPU.baseName(), model.idealLoad().gpu(), context);
+        metric.add(ConfigServerMetrics.CLUSTER_AUTOSCALE_PEAK_LOAD_GPU.baseName(), model.peakLoad().gpu(), context);
```
Comment (olaaun) [inline node-repository/src/main/java/com/yahoo/vespa/hosted/provision/autoscale/Autoscaler.java:86]: Use `metric.set` instead

**Comment by bjormel** [comment]
🙄 
Found that @bratseth already added metrics for Ideal load last year in https://github.com/vespa-engine/vespa/pull/27152

**Pull request closed without merging**